CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
Another point is that to support the aforementioned corner case we can't just check if the exe exists but actually need to know if it's "outdated". Right now the Cabal corner case doesn't do any recompilation checking IIRC so it's just always outdated (i.e not prepared) but we might want to add that at some point so we should support that here.

Given that in this case the helper will never be "prepared" how do we expect an API client to react to this? What is the use-case for this even? So far the API contract is "ask me for stuff and I'll make sure I can tell you stuff" so we don't need to worry about any of this but now you're opening up that can of worms.